zebra-chain/src/parameters/network/testnet.rs

```diff
@@ -57,6 +60,17 @@ impl ParameterSubsidy for TestnetParameterSubsidyImpl {
     fn height_for_first_halving(&self) -> Height {
         FIRST_HALVING_TESTNET
     }
+    fn post_blossom_halving_interval(&self) -> block::HeightDiff {
+        POST_BLOSSOM_HALVING_INTERVAL
+    }
+
+    fn pre_blossom_halving_interval(&self) -> block::HeightDiff {
+        PRE_BLOSSOM_HALVING_INTERVAL
+    }
+
+    fn funding_stream_address_change_interval(&self) -> HeightDiff {
+        FUNDING_STREAM_ADDRESS_CHANGE_INTERVAL
+    }
```

Comment:
These should be based on the `pre_blossom_halving_interval` config field and the Blossom activation height, and `testnet::ParametersBuilder` should panic or return an error when :
- Setting the halving interval or activation heights after setting the funding streams, or
- Calling `finish()` with a non-default halving interval or Blossom activation height but the default Testnet funding streams (this will affect `new_regtest()`, zcashd has no funding streams on Regtest, we should clear the out the default funding streams in Zebra as well).

Related: zcashd seems to have founders rewards in their Regtest until the `first_halving_height - 1`, we may want to add that in Zebra or remove it in zcashd, we should ask the ECC what they think would be best here.

Reply:
> Related: zcashd seems to have founders rewards in their Regtest until the `first_halving_height - 1`, we may want to add that in Zebra or remove it in zcashd, we should ask the ECC what they think would be best here.

I would say that we probably want to add it to Zebra rather than removing in zcashd

Reply:
Can you do this in a suggestion PR or commit ? It is not fully clear to me how to make it.

Reply:
@mpguerra I was wrong about zcashd having founders' rewards in their Regtest, [it only applies until Canopy and after the genesis block](https://github.com/zcash/zcash/blob/master/src/miner.cpp#L122-L155), and Zebra's Regtest [always activates Canopy at `Height(1)`](https://github.com/ZcashFoundation/zebra/blob/custom-testnet-funding-stream-addr-period-check/zebra-chain/src/parameters/network/testnet.rs#L596), so they already match.

Updated https://github.com/ZcashFoundation/zebra/issues/8434 to include adding a founders' rewards if zcashd hasn't been deprecated before it's being implemented.

Opened suggestion PR #8921 addressing the rest of this comment.